core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestUtils.java

```diff
@@ -165,6 +173,102 @@ private static Response createErrorResponse(S3Exception e) {
     }
   }
 
+  /**
+   * convert the AlluxioStatusException to the HTTP Response.
+   * @param e AlluxioStatusException
+   * @return response Http Response
+   */
+  public static Response createErrorResponse(AlluxioStatusException e) {
+    XmlMapper mapper = new XmlMapper();
+    S3ErrorCode s3ErrorCode;
+    // TODO(WYY): we need to handle more exception in the future.
+    if (e instanceof NotFoundException) {
+      // 404
+      s3ErrorCode = S3ErrorCode.NO_SUCH_KEY;
+    } else if (e instanceof InvalidArgumentException) {
+      // 400
+      s3ErrorCode = S3ErrorCode.INVALID_ARGUMENT;
+    } else if (e instanceof PermissionDeniedException) {
+      // 403
+      s3ErrorCode = S3ErrorCode.ACCESS_DENIED_ERROR;
+    } else if (e instanceof FailedPreconditionException) {
+      // 412
+      s3ErrorCode = S3ErrorCode.PRECONDITION_FAILED;
+    } else {
+      // 500
+      s3ErrorCode = S3ErrorCode.INTERNAL_ERROR;
+    }
+    S3Error errorResponse = new S3Error(e.getMessage(), s3ErrorCode);
+    try {
+      return Response.status(s3ErrorCode.getStatus())
+          .entity(mapper.writeValueAsString(errorResponse)).build();
```

Comment:
Creating a comment in order to link to this elsewhere.

Reply:
> Hey thanks for addressing the feedback, those look good! I have just one lingering request if you could take a look at it. Please message me if you need clarification, I know it's kind of dense.

done